### PR TITLE
ADD: Search UI

### PR DESCRIPTION
**Pull request by co012**

### CODE REVIEW

File: packages/app/features/dashboard/search/search-input-field.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <Icon name="magnify" color="black" size={24} />
+      <TextInput
+        placeholder="Search"
+        placeholderTextColor="black"
+        className="grow ml-1"
+        {...rest}
+        ref={ref}
+      />
+      {showX && (
+        <Pressable onPress={() => onXClick?.()}>
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/search-input-field.tsx:26]:
```suggestion
        <Pressable onPress={() => onXClick?.call()}>
```

File: packages/app/features/dashboard/search/search-input-field.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ...rest
+  }: TextProps & { showX?: boolean; onXClick?: () => void },
+  ref: ForwardedRef<rTextInput>
+) {
+  return (
+    <View className="w-full bg-white rounded-lg px-2 py-1 flex flex-row">
+      <Icon name="magnify" color="black" size={24} />
+      <TextInput
+        placeholder="Search"
+        placeholderTextColor="black"
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/search-input-field.tsx:20]: use colors from theme

File: packages/app/features/dashboard/search/search-input-field.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        className="grow ml-1"
+        {...rest}
+        ref={ref}
+      />
+      {showX && (
+        <Pressable onPress={() => onXClick?.()}>
+          <Icon name="close-circle-outline" color="black" size={24} />
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/search-input-field.tsx:27]: same

File: packages/app/features/dashboard/search/recently-added/list-entry/price.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Text, View } from "app/design-system";
+import Dollar from "app/ui/icons/dollar";
+
+export function Price({
+  className,
+}: {
+  code?: string | null;
+  issuer?: string | null;
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/list-entry/price.tsx:8]: remove it, if they are not used right now

File: packages/app/features/dashboard/search/recently-added/list-entry/index.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { Image } from "react-native";
+import { imageUrlSmall } from "app/utils/entry";
+import Icon from "@expo/vector-icons/MaterialCommunityIcons";
+import { Price } from "app/features/dashboard/search/recently-added/list-entry/price";
+import { FavoriteButton } from "app/features/dashboard/search/recently-added/list-entry/favoriteButton";
+
+export function BeatListEntry({ entry }: { entry: Entry }) {
+  return (
+    <View className="w-full flex flex-row py-2">
+      {/* eslint-disable-next-line jsx-a11y/alt-text */}
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/list-entry/index.tsx:12]: can you disable this rule in the `eslintrc.js`

File: packages/app/features/dashboard/search/recently-added/list-entry/index.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      <View className="ml-2 flex justify-center">
+        <Text className="text-sm font-bold leading-6">{entry.title}</Text>
+        <Text className="text-xs text-neutral-400 leading-6">
+          {entry.artist}
+        </Text>
+      </View>
+      <View className="ml-auto flex flex-row items-center">
+        <Price code={entry.code} issuer={entry.issuer} className="mr-3" />
+        <FavoriteButton size={20} />
+        <Icon name="dots-vertical" size={30} color="white" />
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/list-entry/index.tsx:28]: color from theme

File: packages/app/features/dashboard/search/recently-added/list-entry/favoriteButton.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  size: number;
+  className?: string;
+}) {
+  const [active, setActive] = useState(false);
+  return (
+    <Pressable className={className} onPress={() => setActive(!active)}>
+      {active ? (
+        <Icon name="heart" size={size} color={tw.color("blue")} />
+      ) : (
+        <Icon name="heart-outline" size={size} color="white" />
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/list-entry/favoriteButton.tsx:19]: same

File: packages/app/features/dashboard/search/recently-added/index.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  const [nextPage, setNextPage] = useState(0);
+  const [data, setData] = useState<Entry[]>([]);
+  const { data: queryData, refetch } = useRecentlyAddedQuery({
+    variables: { page: 0 },
+  });
+
+  useEffect(() => {
+    if (queryData && queryData.recentlyAdded) {
+      const recentlyAdded: Entry[] = queryData.recentlyAdded.filter(
+        (entry): entry is Entry => entry !== null
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/index.tsx:17]: add link to stack

File: packages/app/features/dashboard/search/recently-added/index.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      );
+      setData((d) => d.concat(recentlyAdded));
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/index.tsx:19]:
```suggestion
      setData((prev) => prev.concat(recentlyAdded));
```

File: packages/app/features/dashboard/search/recently-added/index.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+      setNextPage((p) => p + 1);
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/index.tsx:20]:
```suggestion
      setNextPage((prev) => prev + 1);
```

File: packages/app/features/dashboard/search/index.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+export function SearchScreen() {
+  const [searchFraze, setSearchFraze] = useState("");
+  const [tab, setTab] = useState<Tabs>("Beats");
+
+  return (
+    <SafeAreaView className="w-full max-w-6xl mx-auto flex-1 flex p-4 bg-blue-dark">
+      <SearchInputField
+        value={searchFraze}
+        onChangeText={setSearchFraze}
+        showX={searchFraze !== ""}
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/index.tsx:17]: use `isEmpty`

File: packages/app/features/dashboard/search/recently-added/index.tsx (hunk 8 above)
Comment (co012) [reply packages/app/features/dashboard/search/recently-added/index.tsx:17]: https://stackoverflow.com/questions/43010737/way-to-tell-typescript-compiler-array-prototype-filter-removes-certain-types-fro

File: packages/app/features/dashboard/search/recently-added/list-entry/price.tsx [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { Text, View } from "app/design-system";
+import Dollar from "app/ui/icons/dollar";
+
+export function Price({ className }: { className?: string }) {
+  const price = 25;
+
+  return (
+    <View className={`flex flex-row items-center ${className}`}>
+      <Dollar size={10} color={"white"} />
```
Comment (jmysliv) [inline packages/app/features/dashboard/search/recently-added/list-entry/price.tsx:9]: use color from tailwind config

**Review by jmysliv — APPROVED** [review]
Looks good! 🚀

**Pull request merged**